### ISSUE TITLE
Ingest causes bg persist topics to become red on all machines

### ISSUE THREAD

**Issue by GuyGuterman** [thomsonreuters/CM-Well#355]
After ingest of several thousand infotons - All persist topics become red. Bg imp-application.log full with following exceptions:
``` 2017-12-24 10:10:18,961 l-serverNode-serverNode-serverNode.actor.default-dispatcher-45] ERROR cmwell.bg.ImpStream                  - ImpStream: PublishParentsSink done with error
java.util.concurrent.ExecutionException: Boxed Error
        at scala.concurrent.impl.Promise$.resolver(Promise.scala:59)
        at scala.concurrent.impl.Promise$.scala$concurrent$impl$Promise$$resolveTry(Promise.scala:51)
        at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:248)
        at scala.concurrent.Promise$class.tryFailure(Promise.scala:112)
        at scala.concurrent.impl.Promise$DefaultPromise.tryFailure(Promise.scala:157)
        at akka.stream.impl.fusing.GraphStages$IgnoreSink$$anon$9.onUpstreamFailure(GraphStages.scala:433)
        at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:485)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:368)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:571)
        at akka.stream.impl.fusing.ActorGraphInterpreter$SimpleBoundaryEvent$class.execute(ActorGraphInterpreter.scala:42)
        at akka.stream.impl.fusing.ActorGraphInterpreter$BatchingActorInputBoundary$OnError.execute(ActorGraphInterpreter.scala:59)
        at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:546)
        at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:728)
        at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:743)
        at akka.actor.Actor$class.aroundReceive(Actor.scala:517)
        at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:653)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:527)
        at akka.actor.ActorCell.invoke(ActorCell.scala:496)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
        at akka.dispatch.Mailbox.run(Mailbox.scala:224)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
        at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: scala.NotImplementedError: an implementation is missing
        at scala.Predef$.$qmark$qmark$qmark(Predef.scala:230)
        at cmwell.bg.Merger.cmwell$bg$Merger$$write_merge(Merger.scala:196)
        at cmwell.bg.Merger$$anonfun$do_merge$1$1.apply(Merger.scala:224)
        at cmwell.bg.Merger$$anonfun$do_merge$1$1.apply(Merger.scala:224)
        at cmwell.bg.Merger.merged$lzycompute$1(Merger.scala:204)
        at cmwell.bg.Merger.cmwell$bg$Merger$$merged$1(Merger.scala:204)
        at cmwell.bg.Merger$$anonfun$ensurePrevUUID$1$1.apply(Merger.scala:205)
        at cmwell.bg.Merger$$anonfun$ensurePrevUUID$1$1.apply(Merger.scala:205)
        at scala.Option.fold(Option.scala:158)
        at cmwell.bg.Merger.ensurePrevUUID$1(Merger.scala:205)
        at cmwell.bg.Merger.do_merge$1(Merger.scala:224)
        at cmwell.bg.Merger.merge_recurse$1(Merger.scala:268)
        at cmwell.bg.Merger.merge(Merger.scala:276)
        at cmwell.bg.ImpStream$$anonfun$55$$anonfun$57.apply(ImpStream.scala:321)
        at cmwell.bg.ImpStream$$anonfun$55$$anonfun$57.apply(ImpStream.scala:321)
        at nl.grons.metrics.scala.Timer.time(Timer.scala:54)
        at cmwell.bg.ImpStream$$anonfun$55.apply(ImpStream.scala:321)
        at cmwell.bg.ImpStream$$anonfun$55.apply(ImpStream.scala:293)
        at akka.stream.impl.fusing.Map$$anon$7.onPush(Ops.scala:47)
        at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:499)
        at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:462)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:368)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:571)
        at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:457)
        ... 14 common frames omitted ```

**Comment by hochgi** [comment]
We added #354 to identify the problem better in logs.
This will help us investigate the cause.

**Comment by hochgi** [comment]
After identified the problem (ingest of un-sanitized dump with `DeletedInfoton`s to `_in` with a powerful auth token that allowed it), this was fixed in #358 by simply not allowing such writes.
